src/config.hpp

```diff
@@ -55,3 +54,10 @@ struct config_struct {
 extern const config_struct config;
 
 extern void init_config();
+
+namespace configs {
+extern void read_config_file(std::unordered_map<std::string, std::string>& umap,
+                             std::ifstream& config_stream);
+// Tries to read configs from pre-defined places and returns the path used.
```

Comment:
please let me know if this is not the way you would like the logic to flow i.e. returning the .conf file we actually ended up using.